### PR TITLE
#5290 Fix new transaction data validation

### PR DESCRIPTION
**Pull request by intcooper**
Fixes the validation of a new transaction against the minimum balance and the credit limit for the target account.

- The check is performed only for withdrawals and transfers
- The reference account is specifically the source one; this fixes an issue when the transaction is created from the target account page (the balance was incorrectly the target account's).

- - -
This change is [Reviewable](https://reviewable.io/reviews/moneymanagerex/moneymanagerex/5291)

### CODE REVIEW

File: src/transdialog.cpp [hunk 1]
```diff
@@ -702,32 +702,32 @@ bool mmTransDialog::ValidateData()
     //Checking account does not exceed limits
     if (m_new_trx || m_duplicate)
     {
-        bool abort_transaction = false;
-        double new_value = m_trx_data.TRANSAMOUNT;
-
-        if (m_trx_data.TRANSCODE == Model_Checking::all_type()[Model_Checking::WITHDRAWAL])
+        if ((m_trx_data.TRANSCODE == Model_Checking::all_type()[Model_Checking::WITHDRAWAL]) ||
```
Comment (vomikan) [inline src/transdialog.cpp:705]: m_trx_data.TRANSCODE != Model_Checking::all_type()[Model_Checking::DEPOSIT]

File: src/transdialog.cpp [hunk 2]
```diff
@@ -702,32 +702,32 @@ bool mmTransDialog::ValidateData()
...
+            (m_trx_data.TRANSCODE == Model_Checking::all_type()[Model_Checking::TRANSFER]))
         {
-            new_value *= -1;
-        }
+            const auto fromAccount{ Model_Account::instance().get(m_trx_data.ACCOUNTID) };
```
Comment (whalley) [inline src/transdialog.cpp:708]: The fromAccount is already available in `account` (see line 594)

Comment (whalley) [inline src/transdialog.cpp:708]: We don't need this as we already have the account pointer in `account`.

Comment (intcooper) [reply src/transdialog.cpp:708]: I agree. I will just use `account`.

File: src/transdialog.cpp (hunk 1 above)
Comment (intcooper) [reply src/transdialog.cpp:705]: I figured that Model_Checking defines constants for the types; see my last commit.

**Pull request merged**